Source/Dafny/DafnyAst.cs

```diff
@@ -4698,35 +4732,49 @@ public enum ES { NotYetComputed, Never, ConsultTypeArguments }
 
   public class TupleTypeDecl : IndDatatypeDecl
   {
-    public readonly int Dims;
+    // TODO(fpoli): `IsGhostArguments` should be read as "it's the isGhost information for the arguments", but it
+    // might sound grammatically incorrect. An alternative is `ArgumentsGhostness`.
+    public readonly List<bool> IsGhostArguments;
```

Comment:
Is the `IsGhostArguments` name fine? I don't have a strong preference.

Reply:
How about `ArgumentGhostness`?

Reply:
`ArgumentGhostness`.